src/main/java/com/base64/d/StudyGuideChecker.java

```diff
@@ -8,127 +8,120 @@
 
 import java.io.IOException;
 import java.util.Base64;
-import java.util.Date;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class StudyGuideChecker {
-    int count = 0;
-    public int totalLinksFound = 0;
-    String url3 = "https://content.infinitecampus.com/sis/latest/study-guide/ad-hoc-filters-letters-and-data-viewer";
-    String decodedText = "";
-    String iframeSrc = "";
-    String date = new Date().toString();
 
-    String titleOfStudyGuide = "";
-    Set<String> studyGuideLinks = new HashSet<>();
-    Set<String> badLinks = new HashSet<>();
+    public void checkStudyGuides() {
+        int count = 0;
+        int totalLinksFound = 0;
 
-    WriteToFile writeToFile = new WriteToFile();
+        String decodedText = "";
+        String iframeSrc = "";
+        String titleOfStudyGuide = null;
 
-    public void checkStudyGuides() {
+        Set<String> studyGuideLinks = new HashSet<>();
+        Set<String> badLinks = new HashSet<>();
 
-        for (String link2 : AllStudyGuideLinks.ALL_STUDY_GUIDES) {
+        WriteToFile writeToFile = new WriteToFile();
+
+        for (String studyGuideLink : AllStudyGuideLinks.ALL_STUDY_GUIDES) {
+            // Find all links within a Study Guide
             try {
-//
-//          Connect to the study guide url
-                Document doc3 = Jsoup.connect(link2).get();
+                // Connect to the study guide url
+                Document wraperDocument = Jsoup.connect(studyGuideLink).get();
 
-//            grab the iframe
-                Element iframe = doc3.select("iframe").first();
+                // Grab the iframe
+                Element iframe = wraperDocument.select("iframe").first();
 
-//            get iframe scr url
+                // Get iframe src url
                 iframeSrc = iframe.attr("src");
 
-//            connect to the iframe source url
-                Document doc = Jsoup.connect(iframeSrc).get();
+                // Connect to the iframe source url
+                Document studyGuideDocument = Jsoup.connect(iframeSrc).get();
 
-                titleOfStudyGuide = doc3.title();
+                titleOfStudyGuide = wraperDocument.title();
                 System.out.println(titleOfStudyGuide + " " + iframeSrc);
 
-//            get the script tag info
-                Elements scriptTag = doc.getElementsByTag("script");
+                // Get the script tag info
+                Elements scriptTag = studyGuideDocument.getElementsByTag("script");
                 String jsScripts = scriptTag.toString();
 
-//            match any string of characters over 100 in length.
-                String patternString = "(\\w{100,}.+)";
```

Comment:
This regex definitely works, and although it is unlikely to have more than 100 characters together in this class, it would be considered brittle, because some data could be included that would cause problems.
I've changed it to match on the JavaScript variable name, and then reference the double quotes that will be around this object. This would be considered more robust because it is much less likely to fail if the data changes (it's unlikely that a second `window.courseData` object will be created in this class).